inspirehep/dojson/hep/fields/bd01x09x.py

```diff
@@ -130,15 +130,24 @@ def get_value(val):
 
 
 @hep.over('external_system_numbers', '^035..')
-@utils.for_each_value
-@utils.filter_values
 def external_system_numbers(self, key, value):
     """System Control Number."""
-    return {
-        'value': value.get('a'),
-        'institute': value.get('9'),
-        'obsolete': value.get('z'),
-    }
+    def get_value(value):
+        return {
+            'value': value.get('a'),
+            'institute': value.get('9'),
+            'obsolete': bool(value.get('z')),
+        }
+
+    external_system_numbers = self.get('external_system_numbers', [])
+
+    if isinstance(value, list):
```

Comment:
To avoid duplication of errors I think we can avoid the `isinstance` pattern by just using `utils.force_list` and always treat as list. WDYT?